### PR TITLE
Regression: Discussions were not showing on Tab Bar

### LINKED ISSUE #14040: [BUG] Discussions are not showing on Tab Bar
### Description:

Discussion for channel are not showing.

### Steps to reproduce:

1. click 3 dots on channel top right
2. click discussions

### Expected behavior:

Discussions should appear

### Actual behavior:

It's empty

### Server Setup Information:

- Version of Rocket.Chat Server: 1.0.0-develop
- Operating System: MacOS
- Deployment Method: Meteor
- Number of Running Instances: 1
- DB Replicaset Oplog: Enabled
- NodeJS Version: 8.11.4
- MongoDB Version: 3.6

### Additional context

This is an important issue. On it, fixing it ASAP

### PR DESCRIPTION
**Pull request by knrt10**
The problem was that collection was initially not loading, so `count()` function was not working. Another main problem was in `Template.subscriptionsReady` beacause the subscription was no longer ready, so the whole template was being un-rendered and re-rendered, for a relatively small change that can be optimistically predicted. Fixed it now

Closes #14040

cc @tassoevan @ggazzo would you please review

### CODE REVIEW

File: app/discussion/client/views/DiscussionTabbar.js [hunk 1]
```diff
@@ -24,24 +28,23 @@ Template.discussionsTabbar.helpers({
...
-	});
 	this.hasMore = new ReactiveVar(true);
 	this.limit = new ReactiveVar(50);
-	return this.autorun(() => {
+	this.cursor = new ReactiveVar([]);
+	this.autorun(() => {
 		const data = Template.currentData();
-		return this.subscribe('discussionsOfRoom', data.rid, this.limit.get(), () => {
-			if (DiscussionOfRoom.find({
+		const handle = Meteor.subscribe('discussionsOfRoom', data.rid, this.limit.get());
```
Comment (ggazzo) [inline app/discussion/client/views/DiscussionTabbar.js]: if you are using `Meteor.subscribe` you should include a `sub.stop()` `this.subscribe` make the stop on `onDestroy` to you.

File: app/discussion/client/views/DiscussionTabbar.js [hunk 2]
```diff
@@ -24,24 +28,23 @@ Template.discussionsTabbar.helpers({
...
+		const isReady = handle.ready();
+		// Wait for collection to be ready
+		if (isReady) {
+			const discussions = DiscussionOfRoom.find({
 				rid: data.rid,
-			}).count() < this.limit.get()) {
+			}).fetch();
+			this.cursor.set(discussions);
```
Comment (ggazzo) [inline app/discussion/client/views/DiscussionTabbar.js]: its now how cursor works, if you use a fetch()... its the data itself and not a cursor (:

File: app/discussion/client/views/DiscussionTabbar.js (hunk 1 above)
Comment (knrt10) [reply app/discussion/client/views/DiscussionTabbar.js]: updated

File: app/discussion/client/views/DiscussionTabbar.js (hunk 2 above)
Comment (knrt10) [reply app/discussion/client/views/DiscussionTabbar.js]: So I should try with cursors only?

Comment (knrt10) [reply app/discussion/client/views/DiscussionTabbar.js]: i get it cause we don't want to load all of them into your server's RAM. Let me fix with cursors

**Comment by knrt10** [comment]
@ggazzo updated the PR with cursors.

**Pull request merged**